### ISSUE TITLE
RFE: add an option to update roles and collections from requirements.yml like AWX / automation controller?

### ISSUE THREAD

**Issue by bonnevil** [ansible/ansible-navigator#572]
##### ISSUE TYPE
 - Add an option or options to **ansible-navigator run** that allow it to use the `requirements.yml` or `collections/requirements.yml` files in the playbook's directory to automatically install roles and collections in the EE.

##### SUMMARY
AWX / automation controller / Ansible Tower have a feature (which I believe is turned on by default in Settings -> Jobs) that causes the automation controller to automatically install roles and collections specified in the playbook's directory by `requirements.{yml,yaml}` or `collections/requirements.{yml,yaml}` files.  The playbook that automation controller uses to do this can be seen at https://github.com/ansible/awx/blob/devel/awx/playbooks/project_update.yml and the task involved is at line 196 at the time I'm writing this.  This allows developers to avoid having to check the same collection into multiple projects in Git.

It would be really nice if **ansible-navigator** also had options to honor these `requirements.yml` files and use **ansible-galaxy** to automatically install roles and collections inside the EE.  It would be convenient for developers and would parallel the behavior of AWX / automation controller.

**Comment by bonnevil** [comment]
In the short run, I think our solution for the course should be one of the following, and it **will** vary from hands-on activity to hands-on activity:

1) Set up `.gitignore` to ignore `./collections/ansible_collections`
2) **OR** set up `.gitignore` to ignore specific collections in that location

**AND**

Pull collections from automation hub with `ansible-galaxy install -r collections/requirements.yml` prior to running `ansible-navigator`, but unless the collection is not in automation hub but one we're writing for just this project (possible for the first part of ch09), do **not** commit the collection directly in the project directory in Git because automation controller will automatically pull it from Git when it launches the playbook as a job.

**Comment by cidrblock** [comment]
Good idea, this would probably be exposed as an additional subcommand which ran galaxy for the user.  Will be looked at closer later this year.

**Comment by one-t** [comment]
👍 - this will make the developer workflow moving between Navigator -> Controller much smoother.

**Comment by cidrblock** [comment]
Will need to check what the default behaviour of controller/AWX is, I'm assuming it is "missing", being it will collect anything not present, but not upgrade anything that is...

I'm eager to do this one after 2.0 is out the door :)

**Comment by jdptechnc** [comment]
I would like to see this enhancement as well!

**Comment by Hapshanko** [comment]
Is there an update on this?
Currently manually installing our requirements.yml SCM roles, and then mounting them into the execution environment, so we could really use this solution instead.

**Comment by michten** [comment]
I just faced that it is missing in navigator 2.1... As the previous ones wrote, we need this, as this was whole point of navigator to use it for playbooks deployment for AAP.
